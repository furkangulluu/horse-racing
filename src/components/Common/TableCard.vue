<template>
  <div class="table-card__card border-2">
    <div class="horse-list__card__header border-b p-2 bg-gray-100">
      <span>{{ title }}</span>
    </div>
    <table class="table-card__table w-full">
      <thead>
        <slot name="thead"></slot>
      </thead>
      <tbody>
        <slot name="tbody"></slot>
      </tbody>
    </table>
  </div>
</template>

<script setup lang="ts">
const props = defineProps({
  title: {
    type: String,
    default: "",
  },
});
</script>
