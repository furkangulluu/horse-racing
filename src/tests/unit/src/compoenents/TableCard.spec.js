import { describe, expect, test } from "vitest";
import { mount } from "@vue/test-utils";
import TableCard from "@/components/Common/TableCard.vue";

describe("TableCard.vue", () => {
  let wrapper;
  test("renders the correct title", () => {
    wrapper = mount(TableCard, {
      props: {
        title: "Test Title",
      },
    });

    const titleSpan = wrapper.find(".horse-list__card__header span");
    expect(titleSpan.text()).toBe("Test Title");
  });

  test("is correctly rendered", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("has correct structure and classes", () => {
    const wrapper = mount(TableCard);

    expect(wrapper.classes()).toContain("table-card__card");
    expect(wrapper.find("div.horse-list__card__header").exists()).toBe(true);
    expect(wrapper.find("table").exists()).toBe(true);
    expect(wrapper.find("table").classes()).toContain("table-card__table");
  });
});

describe("TableCard.vue Slot Test", () => {
  test("renders thead and tbody slots correctly", () => {
    const wrapper = mount(TableCard, {
      slots: {
        thead: "<tr><th>Header</th></tr>",
        tbody: "<tr><td>Body</td></tr>",
      },
    });

    const theadContent = wrapper.find("thead");
    const tbodyContent = wrapper.find("tbody");

    expect(theadContent.html()).toContain("Header");
    expect(tbodyContent.html()).toContain("Body");
  });

  test("renders no content when slots are empty", () => {
    const wrapper = mount(TableCard);

    const theadContent = wrapper.find("thead");
    const tbodyContent = wrapper.find("tbody");

    expect(theadContent.html()).toBe("<thead></thead>");
    expect(tbodyContent.html()).toBe("<tbody></tbody>");
  });
});
