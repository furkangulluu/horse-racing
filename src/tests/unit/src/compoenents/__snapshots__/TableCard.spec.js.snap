// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ActionButton.vue > is correctly rendered 1`] = `"<div class="action-button flex justify-center items-center py-2 px-6 h-10 rounded-lg cursor-pointer ml-3 transition-all bg-off-white hover:bg-gray-200 text-black">Button</div>"`;

exports[`TableCard.vue > is correctly rendered 1`] = `
"<div class="table-card__card border-2">
  <div class="horse-list__card__header border-b p-2 bg-gray-100"><span>Test Title</span></div>
  <table class="table-card__table w-full">
    <thead></thead>
    <tbody></tbody>
  </table>
</div>"
`;
